SWLOR.Game.Server/Native/ResolveAttackRoll.cs

```diff
@@ -24,7 +28,366 @@ public static void RegisterHook()
+                {
+                    // Force powers or point blank shot feat make close range an advantage.
+                    if (attackType == 3 || attacker.m_pStats.HasFeat((ushort)FeatType.PointBlankShot) == 1)
+                    {
+                        modifiers += 5;
+                    }
+                    else
+                    {
+                        modifiers -= 20;
+                    }
+                }
+                else if (range > 40.0f)
+                {
+                    modifiers -= 20;
+                }
+                else if (range > 30.0f)
+                {
+                    modifiers -= 10;
+                }
+                else if (range > 20.0f)
+                {
+                    modifiers = -5;
+                }
+                
+            }
+
+            // Attacking from behind.  Does not apply to Force attacks.
+            // Vectors are X=-1 to +1, Y=-1 to +1, Z=0.  
+            // If the absolute difference between the two vectors is less than 0.5, treat as a backstab.
+            Vector attFacing = attacker.m_vOrientation;
+            Vector defFacing = defender.m_vOrientation;
+
+            if (attackType != (uint)AttackType.Spirit && 
+                (Math.Abs(attFacing.x - defFacing.x) + Math.Abs(attFacing.y-defFacing.y) < 0.5))
+            {
+                Log.Write(LogGroup.Attack, "Backstab!  Attacker facing: " + attFacing.x + ", " + attFacing.y + "," + attFacing.z + 
+                                           ", Defender facing: " + defFacing.x + ", " + defFacing.y + "," + defFacing.z);
+                modifiers += 30;
+            }
+
+            // End modifiers
+            //---------------------------------------------------------------------------------------------
+            //---------------------------------------------------------------------------------------------
+            //---------------------------------------------------------------------------------------------
+            int roll = Service.Random.Next(1, 100);
+            int bonus = 10 * attackAttribute - 10 * defendAttribute + modifiers;
+            // Update the hit roll and modifier to give player feedback.  
+            // Hit roll is 1-100
+            // Modifier is the delta between the attacker & target attributes, updated for any modifiers.
+            // NWN can only display numbers up to 32.  So divide by 4 so that the numbers make sense.
+            Log.Write(LogGroup.Attack, "Roll: " + roll.ToString() + ", bonus: " + bonus.ToString());
+            pAttackData.m_nToHitRoll = (byte) (roll / 4);
+            pAttackData.m_nToHitMod = (byte) (bonus / 4);
+            int result = roll - 50 + bonus;
+
+            if (result < 0)
+            { 
+                // Miss
+                pAttackData.m_nAttackResult = 4;
+                pAttackData.m_nMissedBy = (byte) Math.Abs(result); // Dunno if this is needed by anything, but filling it out in case.
```

Comment:
MissedBy would indicate to me that it's the attacker who missed. I assume if you're casting to a byte here it's not a uint type referencing a game object. We should try to track down what this is, if possible. Makes me nervous setting a value but not knowing what it touches

Reply:
In the original code it's set to 1 if someone rolls a 1, or the difference between the attack roll and the AC if the roll was not 1 and the attack missed.  So I tried to give it equivalent meaning in this code.

It might be used for animation purposes to look for attacks that were near misses?  Or maybe was just created as a hook and never used.  I'd also feel more comfortable knowing what it's for, but not sure how to find out!